### PR TITLE
SNOW-526816: Add long running test script

### PR DESCRIPTION
**Pull request by sfc-gh-yixie**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-526816

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Use the python scripts with command line to repeatedly run DataFrame operations for long time (hours, or days)
For instance, run the test with sql simplifier and 50 columns for 3 days (259200 seconds) and output CPU and Memory usage every 60 seconds.
```cmd
python long_runner.py --duration 259200 --res_interval 60 --log_folder "results/simplifier" --columns 50 --simplify
```

### CODE REVIEW

File: tests/perf/long_runner.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    df = df.with_column(
+        "window_value", avg(df[1]).over(Window.partition_by(df[2]).order_by(df[3]))
+    )
+    with df.cache_result() as cached_df:
+        union_df = cached_df.union(df)
+        join_df = union_df.join(df, union_df[1] == df[1])
+        join_df.collect()
+    group_by_df = df.group_by("a").avg(df[1])
+    group_by_df.collect()
+    for _ in df.to_local_iterator():
```
Comment (sfc-gh-jdu) [inline tests/perf/long_runner.py:65]: we might get the same issue as #496

File: tests/perf/long_runner.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    with_column
+    aggregate
+    window
+    join
+    union
+    collect
+    to_pandas
+    cache_result
+    to_local_iterator
+    to_pandas_batch
```
Comment (sfc-gh-jdu) [inline tests/perf/long_runner.py:45]: should we add 1) async api 2) save_as_table 3) write_pandas, which seem important and some users complained before?

**Comment by sfc-gh-jdu** [comment]
Also, should we create a jenkins test to running it regularly?

**Comment by sfc-gh-sfan** [comment]
This script looks good. Second @sfc-gh-jdu's point, do we want to create a daily Jenkins test for this?

**Comment by sfc-gh-yixie** [comment]
> Also, should we create a jenkins test to running it regularly?

I created a JIRA to create a jenkins job after GA.

**Review by sfc-gh-jdu — APPROVED** [review]

**Review by sfc-gh-sfan — APPROVED** [review]

**Pull request merged**